app/src/components/api-store.js

```diff
@@ -0,0 +1,58 @@
+
+const ApiRunner = async (method, endpoint, payload, callback) => {
+  const domain = 'http://127.0.0.1:5000/api'
+
+  const RequestInfo = {
+    method
+  }
+
+  if (method === 'POST' && 'body' in payload) {
+    RequestInfo.body = payload.body
+  }
+
+  if ('headers' in payload) {
+    RequestInfo.headers = payload.headers
+  }
+
+  return await fetch(domain+endpoint, RequestInfo).then(callback).catch()
+}
+
+const ApiStore = () => (
+  {
+    getToken: async ({email, password}) => {
+      const payload = {
+        headers: new Headers({
+          'Authorization': 'Basic ' + btoa(email +":"+password)
+        })
+      }
+
+      const callback = (res) => {
+        if (res.ok) {
+          const result = res.text()
+          return result
+        }
+      }
+
+      return await ApiRunner('POST', "/tokens", payload, callback)
+    },
```

Comment:
getToken is the only route that will use the `Authorization:  Basic...` Header. Anything else that requires authorization will use `Authorization: Bearer ...`